crates/common/src/tree.rs

```diff
@@ -436,12 +463,12 @@ where
                 let key_hash = KeyHash::with::<Hasher>(hashed_id);
 
                 // hashchain should not already exist
-                if self.get(key_hash)?.is_ok() {
+                let HashchainResponse::NotFound(_) = self.get(key_hash)? else {
```

Comment:
Here is a case where I actually like the let else syntax, maybe because it's not assigning a new variable from it